packages/core/src/tracing/idletransaction.ts

```diff
@@ -158,15 +158,24 @@ export class IdleTransaction extends Transaction {
             logger.log('[Tracing] cancelling span since transaction ended early', JSON.stringify(span, undefined, 2));
         }
 
-        const keepSpan = span.startTimestamp < endTimestamp;
-        if (!keepSpan) {
-          __DEBUG_BUILD__ &&
-            logger.log(
-              '[Tracing] discarding Span since it happened after Transaction was finished',
-              JSON.stringify(span, undefined, 2),
-            );
+        const spanStartedBeforeTransactionFinish = span.startTimestamp < endTimestamp;
+
+        const timeoutWithDelta = (this._finalTimeout + this._idleTimeout) / 1000;
+
+        const transactionDidNotExceedTimeout = endTimestamp - this.startTimestamp < timeoutWithDelta;
```

Comment:
In case the Tx exceeds it's finalTimeout I think this bool will cause all spans to be removed.